src/DependencyInjection/Compiler/ExtensionCompilerPass.php

```diff
@@ -215,6 +235,30 @@ protected function hasTrait(\ReflectionClass $class, $traitName)
         return $this->hasTrait($parentClass, $traitName);
     }
 
+    /**
+     * @phpstan-param class-string $class
+     * @phpstan-param class-string $subject
+     */
+    private function isSubtypeOf(string $type, string $class, string $subject): bool
+    {
+        $classReflection = new \ReflectionClass($class);
+
+        switch ($type) {
+            case 'instanceof':
+                $subjectReflection = new \ReflectionClass($subject);
+
+                return $classReflection->isSubclassOf($subject) || $subjectReflection->getName() === $classReflection->getName();
+            case 'implements':
+                return $classReflection->implementsInterface($subject);
+            case 'extends':
+                return $classReflection->isSubclassOf($subject);
+            case 'uses':
+                return $this->hasTrait($classReflection, $subject);
+        }
+
+        return false;
```

Comment:
In case `$type` could be optional:
```suggestion
        return is_a($class, $subject, true);
```

Reply:
This is not the purpose of the method, also type is a required field

Reply:
I'm aware that `$type` argument is currently required, that's why I left this suggestion indicating this change could be possible if we make it optional.